experimental/saitypesextensions.h

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2018 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saitypesextensions.h
+ *
+ * @brief   This module defines type extensions of the Switch Abstraction Interface (SAI)
+ */
+
+#ifndef __SAITYPESEXTENSIONS_H_
+#define __SAITYPESEXTENSIONS_H_
+
+#include <saitypes.h>
+
+/**
+ * @brief SAI object type extensions
+ *
+ * @flags
```

Comment:
Do we need the flags tag ? Don't we have a contiguous range ?

Reply:
we have, but @flags is enum bound, and this enum is not starting from "0" but from last enum of corresponding sai object type enum

Reply:
OK, thanks